### ISSUE TITLE
The fix of 'vue/multi-spaces' causing 'vue/no-parsing-error'

### ISSUE THREAD

**Issue by emosheeep** [rashfael/eslint-plugin-vue-pug#17]
Hi, there! Sorry for bothering. I encountered a strange problem that when I executed auto-fix of 'vue/multi-spaces'. It broke my logic and caused unexpected error.

Here is the detail：

![屏幕录制2022-10-22 00 06 11](https://user-images.githubusercontent.com/31957758/197240904-46def59e-a3f8-4f9a-87b2-df0613ad1bb8.gif)

What made me confused is that it worked well when i made some changes such as remove the 'src' property of img/video or remove the style code at the bottom of the file. Both of which don't seem to be connected at all!

And here is the [reproduction](https://github.com/emosheeep/eslint-pug-error-reproduction/blob/main/src/App.vue)

Looking forward to your reply!

**Comment by rashfael** [comment]
Hi, sorry for taking so long to respond.
I can't reproduce a `vue/no-multi-spaces` error with neither of your examples.
I tested your examples with a full install and I tested the rule in isolation by [adding a test case here]( https://github.com/rashfael/eslint-plugin-vue-pug/blob/761cbd90fff69c16428974627577317545a4533b/tests/eslint-plugin-vue/rules/no-multi-spaces.js#L55) and running
```
npx mocha -b tests/eslint-plugin-vue/rules/no-multi-spaces.js
```

None of which gave me any issues like you described them.

Which version of `eslint` and `eslint-plugin-vue` are you using?

**Comment by emosheeep** [comment]
@rashfael  I put a reproduction link in the description, did you get the problem with it?  Both of these two key dependencies are the latest version.

**Comment by rashfael** [comment]
I had another look at your repro, which when I run it exactly like it is, gives me the behaviour you describe. But when I remove or add lines in `script` or `style´ the warning disappears, which is extremely weird.

**Comment by rashfael** [comment]
I did some console.log debugging with the short example from @gilles-crealp and the pug template tokenizer returns the same tokens. But the tokens arriving at the rule here https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/no-multi-spaces.js#L66 are different.

<details>
<summary>With error</summary>

```js
[
  {
    type: 'HTMLTagOpen',
    range: [ 0, 9 ],
    loc: { start: [Object], end: [Object] },
    value: 'template'
  },
  {
    type: 'HTMLIdentifier',
    range: [ 10, 14 ],
    loc: { start: [Object], end: [Object] },
    value: 'lang'
  },
  {
    type: 'HTMLAssociation',
    range: [ 14, 15 ],
    loc: { start: [Object], end: [Object] },
    value: ''
  },
  {
    type: 'HTMLLiteral',
    range: [ 15, 20 ],
    loc: { start: [Object], end: [Object] },
    value: 'pug'
  },
  {
    type: 'HTMLTagClose',
    range: [ 20, 21 ],
    loc: { start: [Object], end: [Object] },
    value: ''
  },
  {
    type: 'PugNewline',
    value: '\n',
    loc: { start: [Object], end: [Object] },
    range: [ 21, 22 ]
  },
  {
    type: 'PugTag',
    value: 'p',
    loc: { start: [Object], end: [Object] },
    range: [ 22, 23 ]
  },
  {
    type: 'VExpressionStart',
    value: '{{',
    loc: { start: [Object], end: [Object] },
    range: [ 24, 26 ]
  },
  Token {
    type: 'Identifier',
    value: 'a',
    start: 27,
    end: 28,
    loc: SourceLocation { start: [Object], end: [Object] },
    range: [ 27, 28 ]
  },
  Token {
    type: 'Identifier',
    value: 'a',
    start: 35,
    end: 36,
    loc: SourceLocation { start: [Object], end: [Object] },
    range: [ 35, 36 ]
  },
  {
    type: 'VExpressionEnd',
    value: '}}',
    loc: { start: [Object], end: [Object] },
    range: [ 37, 39 ]
  },
  {
    type: 'PugNewline',
    value: '\n',
    loc: { start: [Object], end: [Object] },
    range: [ 39, 40 ]
  },
  {
    type: 'HTMLEndTagOpen',
    range: [ 40, 50 ],
    loc: { start: [Object], end: [Object] },
    value: 'template'
  },
  {
    type: 'HTMLTagClose',
    range: [ 50, 51 ],
    loc: { start: [Object], end: [Object] },
    value: ''
  }
]
```
</details>

<details>
<summary>Without error</summary>

```js
[
  {
    type: 'HTMLTagOpen',
    range: [ 0, 9 ],
    loc: { start: [Object], end: [Object] },
    value: 'template'
  },
  {
    type: 'HTMLIdentifier',
    range: [ 10, 14 ],
    loc: { start: [Object], end: [Object] },
    value: 'lang'
  },
  {
    type: 'HTMLAssociation',
    range: [ 14, 15 ],
    loc: { start: [Object], end: [Object] },
    value: ''
  },
  {
    type: 'HTMLLiteral',
    range: [ 15, 20 ],
    loc: { start: [Object], end: [Object] },
    value: 'pug'
  },
  {
    type: 'HTMLTagClose',
    range: [ 20, 21 ],
    loc: { start: [Object], end: [Object] },
    value: ''
  },
  {
    type: 'PugNewline',
    value: '\n',
    loc: { start: [Object], end: [Object] },
    range: [ 21, 22 ]
  },
  {
    type: 'PugTag',
    value: 'p',
    loc: { start: [Object], end: [Object] },
    range: [ 22, 23 ]
  },
  {
    type: 'VExpressionStart',
    value: '{{',
    loc: { start: [Object], end: [Object] },
    range: [ 24, 26 ]
  },
  Token {
    type: 'Identifier',
    value: 'a',
    start: 27,
    end: 28,
    loc: SourceLocation { start: [Object], end: [Object] },
    range: [ 27, 28 ]
  },
  {
    type: 'VExpressionEnd',
    value: '}}',
    loc: { start: [Object], end: [Object] },
    range: [ 29, 31 ]
  },
  {
    type: 'PugText',
    value: ' ',
    loc: { start: [Object], end: [Object] },
    range: [ 30, 31 ]
  },
  {
    type: 'VExpressionStart',
    value: '{{',
    loc: { start: [Object], end: [Object] },
    range: [ 32, 34 ]
  },
  Token {
    type: 'Identifier',
    value: 'a',
    start: 35,
    end: 36,
    loc: SourceLocation { start: [Object], end: [Object] },
    range: [ 35, 36 ]
  },
  {
    type: 'VExpressionEnd',
    value: '}}',
    loc: { start: [Object], end: [Object] },
    range: [ 37, 39 ]
  },
  {
    type: 'PugNewline',
    value: '\n',
    loc: { start: [Object], end: [Object] },
    range: [ 39, 40 ]
  },
  {
    type: 'HTMLEndTagOpen',
    range: [ 40, 50 ],
    loc: { start: [Object], end: [Object] },
    value: 'template'
  },
  {
    type: 'HTMLTagClose',
    range: [ 50, 51 ],
    loc: { start: [Object], end: [Object] },
    value: ''
  }
]
```
</details>

*Something* somewhere between the tokenizer (my code) and the rule ate the tokens in the range 28-35, based on … other content not in template?

**Comment by emosheeep** [comment]
28-35 were `}} {{` so this problem might belong to eslint-plugin-vue？Whether it is necessary to open a new issue？

**Comment by gilles-crealp** [comment]
> I had another look at your repro, which when I run it exactly like it is, gives me the behaviour you describe. But when I remove or add lines in `script` or `style´ the warning disappears, which is extremely weird.

Yes, I have the same behavior. Very strange…

**Comment by rashfael** [comment]
Ok, staring at the tokenizer output for way too long:
<details>
<summary>tokenizer output</summary>

```js
[
  {
    type: 'PugNewline',
    value: '\n',
    loc: { start: [Object], end: [Object] },
    range: [ 21, 22 ]
  },
  {
    type: 'PugTag',
    value: 'p',
    loc: { start: [Object], end: [Object] },
    range: [ 22, 23 ]
  },
  {
    type: 'VExpressionStart',
    value: '{{',
    loc: { start: [Object], end: [Object] },
    range: [ 24, 26 ]
  },
  {
    type: 'PugText',
    value: ' a ',
    loc: { start: [Object], end: [Object] },
    range: [ 26, 29 ]
  },
  {
    type: 'VExpressionEnd',
    value: '}}',
    loc: { start: [Object], end: [Object] },
    range: [ 29, 31 ]
  },
  {
    type: 'PugText',
    value: ' ',
    loc: { start: [Object], end: [Object] },
    range: [ 30, 31 ]
  },
  {
    type: 'VExpressionStart',
    value: '{{',
    loc: { start: [Object], end: [Object] },
    range: [ 32, 34 ]
  },
  {
    type: 'PugText',
    value: ' a ',
    loc: { start: [Object], end: [Object] },
    range: [ 34, 29 ]
  },
  {
    type: 'VExpressionEnd',
    value: '}}',
    loc: { start: [Object], end: [Object] },
    range: [ 37, 39 ]
  },
  {
    type: 'PugNewline',
    value: '\n',
    loc: { start: [Object], end: [Object] },
    range: [ 39, 40 ]
  }
]
```
</details>

I noticed this: `range: [ 34, 29 ]`, which is definitely wrong. I will investigate further.

**Comment by emosheeep** [comment]
Holy shit, it’s too hard to find. Appreciate your work. 🐂🍺

**Comment by gilles-crealp** [comment]
> I noticed this: `range: [ 34, 29 ]`, which is definitely wrong.

Good catch!

**Comment by rashfael** [comment]
I found the error leading to wrong token locations (actually was part of the test suite but it's easy to miss in the heap of generated token fixtures).
That the warning disappeared when changing other parts of the file seemed to be a red hering, the tokenizer output did not change. Not sure what that is about.
I'll release the fix later today.

**Comment by rashfael** [comment]
Fix has been released in v0.5.5.

**Comment by gilles-crealp** [comment]
@rashfael Thank you for the quick fix! 🚀
